### PR TITLE
Added tests for RowLayout and CardLayout

### PR DESCRIPTION
**Pull request by iojw**
### Contains

3 tests for RowLayout that test the 3 main scenarios: 

- when all the widgets have relative widths specified
- when no widgets have relative widths specified
- when some widgets have relative widths specified

### How to Test
Run RowLayoutTest.java. All tests should pass!

### Outstanding issues before merging

I've made sure to comment on most parts of the code so that it's easier to understand. Do take a look and see if the tests are ok since this is my first time writing unit tests.

P.S. Unit testing isn't as bad as everyone says 😄 Gives you a good understanding of the class.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by iojw** [comment]
I felt that only using CANVAS_WIDTH and CANVAS_HEIGHT constants for some of the tests actually made them harder to read so I went ahead and added variables like WIDTH_1, WIDTH_2 for the more complicated ones. 

I've also added the CardLayoutTest that I've been working on - thought it would be convenient to add it to this PR as well 🙂. It contains only one test which I feel should be sufficient given that CardLayout is a much simpler class compared to RowLayout and ColumnLayout.

Do take a look if the changes are alright 😄.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine-tests/src/test/java/org/terasology/rendering/nui/layouts/CardLayoutTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import org.junit.Test;
+import org.terasology.math.geom.Rect2i;
+import org.terasology.math.geom.Vector2i;
+import org.terasology.rendering.nui.Canvas;
+import org.terasology.rendering.nui.UIWidget;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.*;
```
Comment (rzats) [inline engine-tests/src/test/java/org/terasology/rendering/nui/layouts/CardLayoutTest.java]: No wildcard imports, please! (Here's how to disable them in IDEA: http://stackoverflow.com/a/3348855) [1]

File: engine-tests/src/test/java/org/terasology/rendering/nui/layouts/RowLayoutTest.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+import org.junit.Test;
+import org.terasology.math.geom.Rect2i;
+import org.terasology.math.geom.Vector2i;
+import org.terasology.rendering.nui.Canvas;
+import org.terasology.rendering.nui.UIWidget;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.*;
```
Comment (rzats) [inline engine-tests/src/test/java/org/terasology/rendering/nui/layouts/RowLayoutTest.java]: No wildcard imports, please! (Here's how to disable them in IDEA: http://stackoverflow.com/a/3348855) [2]

File: engine-tests/src/test/java/org/terasology/rendering/nui/layouts/CardLayoutTest.java (hunk 1 above)
Comment (iojw) [inline engine-tests/src/test/java/org/terasology/rendering/nui/layouts/CardLayoutTest.java]: Oh man! I swear I disabled those...

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Pull request merged**